pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryConstructorRule.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule.codestyle;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.jaxen.JaxenException;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.ast.ASTBlockStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBody;
+import net.sourceforge.pmd.lang.java.ast.ASTConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTNameList;
+import net.sourceforge.pmd.lang.java.rule.AbstractIgnoredAnnotationRule;
+
+/**
+ * This rule detects when a constructor is not necessary;
+ * i.e., when there is only one constructor, it’s public, has an empty body,
+ * and takes no arguments.
+ */
+public class UnnecessaryConstructorRule extends AbstractIgnoredAnnotationRule {
+
+    private static final String XPATH_EXPRESSION_TO_CONSTRUCTOR
+        = "ClassOrInterfaceBodyDeclaration/ConstructorDeclaration";
+
+    private static final String XPATH_EXPRESSION_TO_ARGLISTEXPRESSION
+        = "ExplicitConstructorInvocation/Arguments/ArgumentList/Expression";
+
+    @Override
+    protected Collection<String> defaultSuppressionAnnotations() {
+        Collection<String> defaultValues = new HashSet<>();
+        defaultValues.add("javax.inject.Inject");
+        return defaultValues;
+    }
+
+    @Override
+    public Object visit(ASTClassOrInterfaceBody node, Object data) {
+
+        List<Node> nodes = getConstructorDeclarationNodes(node);
+
+        //the node has more than one constructor
+        if (nodes.size() != 1) {
+            return super.visit(node, data);
+        }
+
+        ASTConstructorDeclaration cdnode = (ASTConstructorDeclaration) nodes.get(0);
+
+        if (cdnode.isPublic() && !cdnode.hasDescendantMatchingXPath("FormalParameters/*")
+            && !cdnode.hasDescendantOfType(ASTBlockStatement.class) && !cdnode.hasDescendantOfType(ASTNameList.class)
+            && !cdnode.hasDescendantMatchingXPath(XPATH_EXPRESSION_TO_ARGLISTEXPRESSION)
```

Comment:
By contrast, use of XPath here is justified since using bare java would blow up the code with null checking. Once we upgrade to Java 8, monadic processing using Optional could be preferred

Reply:
That was a bug, supposed to be fixed on master by #949. If you're sure that it's the case, please open a ticket, I can't test it right now. Also note that you could have to merge master to get those changes back into your branch, since the PR was merged recently

Reply:
It works fine with this PR;)